custom_components/ha_washdata/learning.py

```
@@ -310,6 +325,58 @@ def _update_model_suggestions(self, now: datetime) -> None:
+    async def _async_run_detection_suggestions(self) -> None:
+        """Run the detection-suggestion pass off the event loop."""
+        try:
+            new_suggestions = await self.hass.async_add_executor_job(
+                self.suggestion_engine.generate_detection_suggestions
+            )
+            if new_suggestions:
+                self._apply_suggestions_and_notify(new_suggestions)
+                self._logger.debug(
+                    "Detection suggestions produced: %s", list(new_suggestions.keys())
+                )
+        except Exception as e:  # pylint: disable=broad-exception-caught
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)